src/Uno.UI/UI/Xaml/UIElement.Pointers.cs

```diff
@@ -577,6 +577,17 @@ private protected void CompleteGesture()
 				GestureRecognizer.CompleteGesture();
 			}
 		}
+
+		private void UpdateRaisedEventFlags(PointerRoutedEventArgs args)
+		{
+			if (!IsGestureRecognizerCreated)
+			{
+				return;
+			}
+
+			var pointerId = args.GetCurrentPoint(this).PointerId;
```

Comment:
This is costly (depending of the platform it could walk the tree / do lot of interop), you should instead do:
```suggestion
			var pointerId = args.Pointer.PointerId;
```